src/node.rs

```diff
@@ -279,39 +294,59 @@ impl<S: StorageExt> SawtoothRaftNode<S> {
 
                 if entry.get_entry_type() == EntryType::EntryNormal {
                     let block_id: BlockId = BlockId::from(Vec::from(entry.get_data()));
-                    debug!("Peer({:?}) committing block {:?}", self.peer_id, block_id);
-                    if match self.leader_state {
-                        Some(LeaderState::Proposing(ref proposed)) => {
-                            &block_id == proposed
-                        },
-                        _ => false,
-                    } {
-                        debug!("Leader({:?}) transitioning to Committing block {:?}", self.peer_id, block_id);
-                        self.leader_state = Some(LeaderState::Committing(block_id.clone()));
-                        self.service.commit_block(block_id.clone()).expect("Failed to commit block");
-                    }
-
-                    if self.follower_state.is_some() {
-                        match self.service.commit_block(block_id.clone()) {
-                            Err(Error::UnknownBlock(_)) => {
-                                debug!(
-                                    "Follower({:?}) tried to commit block before available: {:?}",
-                                    self.peer_id,
-                                    block_id
-                                );
-                                self.follower_state = Some(FollowerState::Committing(block_id));
-                            }
-                            Err(err) => panic!("Failed to commit block {:?}", err),
-                            _ => (),
-                        }
-                    }
+                    self.commit_block(block_id).unwrap_or_else(|block_id| {
+                        // Block can't be committed yet, add to backlog
+                        self.commit_block_backlog.push_back(block_id)
+                    });
                 }
             }
         }
 
         // Advance the Raft
         self.raw_node.advance(ready);
     }
+
+    fn commit_block(&mut self, block_id: BlockId) -> Result<(), BlockId> {
+        // Ensure block is committable (oldest validated block)
+        debug!("VERIFYING - block_id: {:?} in valid blocks: {:?} ", &block_id, self.valid_block_backlog);
+        if !self.valid_block_backlog.contains(&block_id) {
```

Comment:
Should this just check the back of the deque or otherwise check some ordering to satisfy the comment above about oldest first?

Reply:
Ah yes, I will fix that.

Reply:
Updating the comment to reflect the code; block new/valid messages may be received out-of-order, so you cannot rely on the ordering of the `valid_block_backlog`.